app/src/sharedTest/java/org/oppia/android/app/policies/PoliciesFragmentTest.kt

```diff
@@ -0,0 +1,255 @@
+
+  @Inject
+  lateinit var htmlParserFactory: HtmlParser.Factory
+
+  @Inject
+  @field:DefaultResourceBucketName
+  lateinit var resourceBucketName: String
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<PoliciesActivity> = ActivityTestRule(
+    PoliciesActivity::class.java, /* initialTouchMode= */
+    true, /* launchActivity= */
+    false
+  )
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testPrivacyPolicyFragment_checkPrivacyPolicy_isDisplayed() {
+    launch<PoliciesActivity>(createPoliciesActivity(PolicyPage.PRIVACY_POLICY)).use {
+      onView(withId(R.id.policies_description_text_view)).perform(scrollTo())
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testPrivacyPolicyFragment_checkPrivacyPolicyWebLink_isDisplayed() {
+    launch<PoliciesActivity>(createPoliciesActivity(PolicyPage.PRIVACY_POLICY)).use {
+      onView(withId(R.id.policies_web_link_text_view)).perform(scrollTo())
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testPrivacyPolicyFragment_checkTermsOfService_isDisplayed() {
+    launch<PoliciesActivity>(createPoliciesActivity(PolicyPage.TERMS_OF_SERVICE)).use {
+      onView(withId(R.id.policies_description_text_view)).perform(scrollTo())
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testPrivacyPolicyFragment_checkTermsOfServiceWebLink_isDisplayed() {
+    launch<PoliciesActivity>(createPoliciesActivity(PolicyPage.TERMS_OF_SERVICE)).use {
+      onView(withId(R.id.policies_web_link_text_view)).perform(scrollTo())
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testPrivacyPolicyFragment_checkPrivacyPolicy_isCorrectlyParsed() {
+    activityTestRule.launchActivity(createPoliciesActivity(PolicyPage.PRIVACY_POLICY))
+    val privacyPolicyTextView = activityTestRule.activity.findViewById(
+      R.id.policies_description_text_view
+    ) as TextView
+    val htmlParser = htmlParserFactory.create()
+    val htmlResult: Spannable = htmlParser.parseOppiaHtml(
+      getResources().getString(R.string.privacy_policy_content),
+      privacyPolicyTextView
+    )
+    assertThat(privacyPolicyTextView.text.toString()).isEqualTo(htmlResult.toString())
+  }
+
+  @Test
+  fun testPrivacyPolicyFragment_checkTermsOfService_isCorrectlyParsed() {
+    activityTestRule.launchActivity(createPoliciesActivity(PolicyPage.TERMS_OF_SERVICE))
+    val privacyPolicyTextView = activityTestRule.activity.findViewById(
+      R.id.policies_description_text_view
+    ) as TextView
+    val htmlParser = htmlParserFactory.create()
+    val htmlResult: Spannable = htmlParser.parseOppiaHtml(
+      getResources().getString(R.string.terms_of_service_content),
+      privacyPolicyTextView
+    )
+    assertThat(privacyPolicyTextView.text.toString()).isEqualTo(htmlResult.toString())
+  }
+
+  private fun createPoliciesActivity(policies: PolicyPage): Intent {
+    return when (policies) {
```

Comment:
Ditto for this helper--suggest dedicated helpers to open the exact specific activity.

Reply:
Done.